api/envoy/config/listener/v3/quic_config.proto

```diff
@@ -24,11 +24,19 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#protodoc-title: QUIC listener config]
 
 // Configuration specific to the UDP QUIC listener.
-// [#next-free-field: 12]
+// [#next-free-field: 13]
 message QuicProtocolOptions {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.api.v2.listener.QuicProtocolOptions";
 
+  // Configuration for UDP cmsg to pass to listener filters via QuicReceivedPacket.
+  // Both level and type must be specified for cmsg to be saved.
+  message SaveCmsgConfig {
+    optional int32 level = 1;
+
+    optional int32 type = 2;
```

Comment:
Please add a description to each field, including the meaning of the default value (0) in each case. Also, just to make sure, the use of signed integer is intentional, correct?

Reply:
Descriptions added. The default value is now unset. Yes, I used a signed integer even though it seems like negatives are never used, because cmsg_level and cmsg_type in the underlying data structure are signed: https://man7.org/linux/man-pages/man3/cmsg.3.html.